### PR TITLE
Support serialization of url query parameters

### PR DESCRIPTION
**Pull request by mahsashadi**

### CODE REVIEW

File: src/utils/fetch.js [hunk 1]
```diff
@@ -28,13 +28,22 @@
...
-  .filter(k => typeof data[k] !== 'object')
-  .map(k => encodeURIComponent(k) + '=' + encodeURIComponent(data[k]))
-  .join('&');
+
+// /*
+//  * Creates URL request path
+//  */
+export const encodeQueryData = (data) => {
+  const replacer = (k, v)=>(v === undefined ? null : v);
+  const pairs = Object.entries(data).map(([key, val]) => {
```
Comment (andersevenrud) [inline src/utils/fetch.js:37]: Might be better to just filter out any null-ish values before doing the encoding. The backend doesn't really take `null` or `undefined` values anyway.

This way you can just do the following:

```js
const pairs = Object.entries(data)
  .filter(([, val]) => val !== null && val !== undefined)
  .map(([key, val]) => {
    const value = typeof val === 'object' ? JSON.stringify(val) : val;
    return `${encodeURIComponent(key)}=${encodeURIComponent(value)}`;
  });
```

**Review by andersevenrud — CHANGES_REQUESTED** [review]

File: src/utils/fetch.js [hunk 2]
```diff
@@ -28,13 +28,22 @@
...
- * Creates URL request path
- */
-const encodeQueryData = data => Object.keys(data)
-  .filter(k => typeof data[k] !== 'object')
-  .map(k => encodeURIComponent(k) + '=' + encodeURIComponent(data[k]))
-  .join('&');
+
+// /*
+//  * Creates URL request path
+//  */
```
[the remark this answers is not in the document]
Comment (mahsashadi) [reply src/utils/fetch.js:34]: Oops! sorry for that.

**Comment by mahsashadi** [comment]
@andersevenrud changes are done.

**Comment by andersevenrud** [comment]
> filtering client options passed

I'm not I sure I fully understand the issue.

**Review by andersevenrud — APPROVED** [review]

**Pull request merged**